.github/workflows/maven.yaml

```diff
@@ -66,7 +70,16 @@ jobs:
           path: ~/.m2
           key: ${{ runner.os }}-m2-${{ hashFiles('**/pom.xml') }}
           restore-keys: ${{ runner.os }}-m2
-      - name: 'Build Kroxylicious maven project'
+      - name: 'Build Kroxylicious Image on Minikube'
+        run: |
+          GIT_HASH="$(git rev-parse HEAD)"
+          IMAGE_TAG="dev-git-${GIT_HASH}"
+          KROXYLICIOUS_VERSION="$(mvn org.apache.maven.plugins:maven-help-plugin:3.4.0:evaluate -Dexpression=project.version -q -DforceStdout)"
+          KROXYLICIOUS_IMAGE="quay.io/kroxylicious/kroxylicious:${IMAGE_TAG}"
+          minikube image build . -f Dockerfile -t ${KROXYLICIOUS_IMAGE} --build-opt=build-arg=KROXYLICIOUS_VERSION="${KROXYLICIOUS_VERSION}"
```

Comment:
Can we mount the m2/repository from the build server in to save re-downloading everything in the image build?

Reply:
Good idea. I'll give that a go. Either the mount needs to be r/w, or we might need to pre-flight the dependency download somehow.

Reply:
My theory was it would be populated from the github build cache so should in theory not need to download much if anything.

Reply:
you could also potentially orchestrate something like storing a tarball of the dependencies in a github release, and use a Docker build arg to pass it's URL into the build. If the build arg is set, the image build could curl it down and install the maven cache.

This isn't very fast for me locally but I wonder if it will be blazing fast to do within github.